packages/sanity/src/core/studio/packageVersionStatus/PackageVersionStatusProvider.tsx

```diff
@@ -0,0 +1,94 @@
+import {Box, Button, Stack, useToast} from '@sanity/ui'
+import {type ReactNode, useCallback, useEffect, useRef} from 'react'
+import {SANITY_VERSION} from 'sanity'
+import semver from 'semver'
+
+import {hasSanityPackageInImportMap} from '../../environment/hasSanityPackageInImportMap'
+import {useTranslation} from '../../i18n'
+import {checkForLatestVersions} from './checkForLatestVersions'
+
+// How often to to check last timestamp. at 30 min, should fetch new version
+const REFRESH_INTERVAL = 1000 * 60 * 5 // every 5 minutes
+const SHOW_TOAST_FREQUENCY = 1000 * 60 * 30 //half hour
+
+/*
+ * We are currently only checking to see if the sanity module has a new version available.
+ * We can add more packages to this list (e.g., @sanity/vision) if we want to check for more.
+ */
+const currentPackageVersions: Record<string, string> = {
+  sanity: SANITY_VERSION,
+}
+
+export function PackageVersionStatusProvider({children}: {children: ReactNode}) {
+  const toast = useToast()
+  const {t} = useTranslation()
+  const lastCheckedTimeRef = useRef<number | null>(null)
+
+  const autoUpdatingPackages = hasSanityPackageInImportMap()
+
+  const showNewPackageAvailableToast = useCallback(() => {
+    const onClick = () => {
+      window.location.reload()
+    }
+
+    toast.push({
+      id: 'new-package-available',
+      title: t('package-version.new-package-available.title'),
+      description: (
+        <Stack space={2} paddingBottom={2}>
+          <Box>{t('package-version.new-package-available.description')}</Box>
+          <Box>
+            <Button
+              aria-label={t('package-version.new-package-available.reload-button')}
+              onClick={onClick}
+              text={t('package-version.new-package-available.reload-button')}
+              tone={'primary'}
+            />
+          </Box>
+        </Stack>
+      ),
+      closable: true,
+      status: 'info',
+      //covering for some delays, etc. because of the toast ID, we should never see this twice
+      duration: SHOW_TOAST_FREQUENCY * 2,
```

Comment:
is there an option for like an infinite duration? it seems like we just want this up indefinitely until the user closes this manually and it seems like the toast ID prevents double of the same message. would that be better?

Reply:
we have precedent!

https://github.com/sanity-io/sanity/blob/4b89dd555e22d91372119c33de39530c2b01f979/packages/sanity/src/structure/panes/document/documentPanel/documentViews/useConditionalToast.ts#L13

in fact, it may just make sense to reuse the useConditionalToast util here...